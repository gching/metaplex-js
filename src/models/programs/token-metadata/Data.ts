import { BorshSchema } from '../../utils/types';
import { Creator } from './Creator';

export class Data {
  static SCHEMA: BorshSchema = new Map([
    [
      Data,
      {
        kind: 'struct',
        fields: [
          ['name', 'string'],
          ['symbol', 'string'],
          ['uri', 'string'],
          ['sellerFeeBasisPoints', 'u16'],
          ['creators', { kind: 'option', type: [Creator] }],
        ],
      },
    ],
    ...Creator.SCHEMA,
  ]);

  name: string;
  symbol: string;
  uri: string;
  sellerFeeBasisPoints: number;
  creators: Creator[] | undefined;
  constructor(args: {
    name: string;
    symbol: string;
    uri: string;
    sellerFeeBasisPoints: number;
    creators: Creator[] | undefined;
  }) {
    this.name = args.name;
    this.symbol = args.symbol;
    this.uri = args.uri;
    this.sellerFeeBasisPoints = args.sellerFeeBasisPoints;
    this.creators = args.creators;
  }
}
