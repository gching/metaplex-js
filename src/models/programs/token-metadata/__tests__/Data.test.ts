import { Data } from '../Data';

describe('Data', () => {
  it('should instantiate with the correct values', () => {
    const inst = new Data({
      name: 'Hi',
      uri: 'wow',
      creators: [],
      sellerFeeBasisPoints: 100,
      symbol: 'DSD',
    });

    expect(inst).toMatchSnapshot();
  });
});
