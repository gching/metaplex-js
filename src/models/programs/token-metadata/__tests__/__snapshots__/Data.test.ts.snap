// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data should instantiate with the correct values 1`] = `
Data {
  "creators": Array [],
  "name": "Hi",
  "sellerFeeBasisPoints": 100,
  "symbol": "DSD",
  "uri": "wow",
}
`;
